backend/internal/mailservice/sendgrid_mail_service.go

```diff
@@ -30,27 +31,28 @@ var notifyPaymentRecordsMail []byte
 //go:embed templates/system_announcement.html
 var systemAnnouncementMail []byte
 
-// MailService struct hods all functionalities of mail service
-type MailService struct {
-	client  *sendgrid.Client
-	metrics *metrics.Metrics
-}
-
 type Attachment struct {
 	FileName string
 	Data     []byte
 }
 
-// NewMailService creates new instance of mail service
-func NewMailService(sendGridKey string, metrics *metrics.Metrics) MailService {
-	return MailService{
+// SendGridMailService provides functionalities for sending emails using SendGrid.
+type SendGridMailService struct {
+	client  *sendgrid.Client
+	metrics *metrics.Metrics
+}
+
+// NewSendGridMailService creates new instance of SendGridMailService
+func NewSendGridMailService(sendGridKey string, metrics *metrics.Metrics) SendGridMailService {
```

Comment:
shouldn't we have a validation here or even in the caller, to have that sendGridKey required?